src/test-apps/happy/lib/WeaveNodeConfigure.py

```diff
@@ -331,7 +331,7 @@ def __init_node(self, node, node_params={}):
             self.logger.error("[localhost] WeaveNodeConfigure: %s" % emsg)
             return
 
-        if self.weave_state.getNodeType(node) in ['ap', 'service']:
+        if self.weave_state.getNodeType(node) in ['ap']:
```

Comment:
Can you explain the nature of this change? Why was it wrong before; how is it correct now?

Reply:
sure, basically this "If" code here will decide if the weave test infra will assign weave address to a node or not. With original code, if a node is service node, it will not assign weave address to the node. Since we need weave address for service node for one of our lwip tunnel test case, so we should remove "service" from the check and service node will get a proper weave address assigned to it.

Reply:
Thanks; that should be clarified in a comment above this block.

Reply:
OK, will add comment for it.